### ISSUE TITLE
PiHole Install Script Breaks Networking

### ISSUE THREAD

**Issue by RCcola1987** [pi-hole/pi-hole#3713]
### Versions
- Pi-hole install script from 8-28-2020
- Pi-hole:  5.1.2
- AdminLTE:  5.1.1
- FTL:  5.2

### Platform

- OS and version: 
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
 
- Platform:  DigitalOcean VPS

### Expected behavior

Script Used: curl -sSL https://install.pi-hole.net | bash

If you cancel the install all changes should be reverted and no system changes made.

### Actual behavior / bug

Script Used: curl -sSL https://install.pi-hole.net | bash

If you use the automated install script but cancel it at the static IP screen by hitting no and then cancel. After system reboot networking is no longer functional. This results in the need to completely reinstall the os.

### Steps to reproduce

1. run script
2. select no at the screen asking if the current system IP should be used as static
3. then on the following screen click cancel
4. reboot

**Comment by dschaper** [comment]
You don't need to completely reinstall the OS. Just edit `/etc/dhcpcd.conf` in the droplet. 

But I agree that IP addressing should only be asked/set/changed on Raspbian, all other distributions should be set by the user.

**Comment by PromoFaux** [comment]
Ah yes, I keep meaning to look at that part. My vote is that we just remove the static IP setting all together, including on Raspbian

**Comment by dschaper** [comment]
I keep meaning to do the same.

I'm in the middle on how to work it. I don't think we need to install `dhcpcd` in any case as Raspbian should already have it installed in the default OS. The question then is to set or not set on Raspbian only. That's more of a behavioral issue, how much hand holding.

**Comment by RCcola1987** [comment]
My question is more broad. If you cancle the installer before it finishes shouldent the system be left in the state before the installer was ran?

**Comment by PromoFaux** [comment]
Presumably you are choosing no here:

![image](https://user-images.githubusercontent.com/1998970/91616388-7cab3400-e97d-11ea-9280-da0ebd3d728e.png)


and then cancel here:

![image](https://user-images.githubusercontent.com/1998970/91616401-82087e80-e97d-11ea-87d2-88d727d12f66.png)

?

**Comment by dschaper** [comment]
>  If you cancle the installer before it finishes

It did finish that section. If we had a better platform to run as an install then we may have more flexibility. Shell scripting just isn't that elegant. I'm assuming you mean the idea of asking all the questions first and then applying the changes after all the questions have been asked?

**Comment by RCcola1987** [comment]
Yes to both. Otherwise it may be better to have provisions in the script to roll back any changes mad up to that point if it is cancled.

**Comment by PromoFaux** [comment]
Starting to explore this here: #3715

**Comment by RCcola1987** [comment]
But will Pull #3715 address the issue of committing system changes before the script has asked all questions and the user has a chance to cancel?

**Comment by dschaper** [comment]
No, that would require a rewrite of the whole installation script.

**Comment by yubiuser** [comment]
Marking this as `Bug:Confirmed` as the broader scope of this issue is:

>  idea of asking all the questions first and then applying the changes after all the questions have been asked

**Comment by yubiuser** [comment]
Should be fixed with https://github.com/pi-hole/pi-hole/pull/5111

We don't touch the hosts DNS settings anymore.

**Comment by yubiuser** [comment]
Pi-hole v6 has been released.
